ipad/ViewControllers/Shift/Model/WatercraftInspectionModel.swift

```diff
@@ -426,63 +432,28 @@ class WatercradftInspectionModel: Object, BaseRealmObject {
         }
     }
     
-    func set(previous dryStorage: Bool) {
-        do {
-            let realm = try Realm()
-            try realm.write {
-                self.previousDryStorage = dryStorage
-                // TODO: Find a way to delete the existing list
-                self.previousWaterBodies = List<PreviousWaterbodyModel>()
-            }
-            
-        } catch let error as NSError {
-            print("** REALM ERROR")
-            print(error)
-        }
-    }
     
-    func set(destination dryStorage: Bool) {
+    func setJournyStatusFlags(dryStorage: Bool, unknown: Bool, commercialManufacturer: Bool, isPrevious: Bool) {
         do {
-            let realm = try Realm()
-            try realm.write {
-                self.destinationDryStorage = dryStorage
-                // TODO: Find a way to delete the existing list
-                self.destinationWaterBodies = List<DestinationWaterbodyModel>()
+            let relam = try Realm()
+            try relam.write {
+                if isPrevious {
+                    self.previousDryStorage = dryStorage
+                    self.unknownPreviousWaterBody = unknown
+                    self.commercialManufacturerAsPrviousWaterBody = commercialManufacturer
+                    // TODO: Find a way to delete the existing list
+                    self.previousWaterBodies =  List<PreviousWaterbodyModel>()
+                } else {
+                    self.destinationDryStorage = dryStorage
+                    self.unknownDestinationWaterBody = unknown
+                    self.commercialManufacturerAsDestinationWaterBody = commercialManufacturer
+                    // TODO: Find a way to delete the existing list
+                    self.destinationWaterBodies = List<DestinationWaterbodyModel>()
```

Comment:
if isPrevious {
   for item in previousWaterBodies {
      RealmRequests.deleteObject(item)
   }
} else {
   for item in destinationWaterBodies {
      RealmRequests.deleteObject(item)
   }
}

Reply:
Thanks! I will update and code and PR accordingly